### PR TITLE
Add `click` / `__linux` constraints to older dask-cuda versions

### PR DESCRIPTION
**Pull request by charlesbluca**
Checklist
* [ ] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

xref: https://github.com/conda-forge/dask-cuda-feedstock/pull/9

cc @conda-forge/dask-cuda

### CODE REVIEW

**Comment by jakirkham** [comment]
Thanks Charles! 😄

Can you please run [`show_diff.py`]( https://github.com/conda-forge/conda-forge-repodata-patches-feedstock/blob/main/recipe/show_diff.py ) and include the results here?

**Comment by charlesbluca** [comment]
Sure!

```
noarch::dask-cuda-0.18.0-pyhd8ed1ab_0.tar.bz2
-    "python >=3.6"
+    "python >=3.6",
+    "click <8.1",
+    "__unix"
noarch::dask-cuda-0.19.0-pyhd8ed1ab_0.tar.bz2
-    "python >=3.6"
+    "python >=3.6",
+    "click <8.1",
+    "__unix"
noarch::dask-cuda-21.10.0-pyhd8ed1ab_0.tar.bz2
-    "python >=3.7"
+    "python >=3.7",
+    "click <8.1",
+    "__unix"
noarch::dask-cuda-21.6.0-pyhd8ed1ab_0.tar.bz2
-    "python >=3.7"
+    "python >=3.7",
+    "click <8.1",
+    "__unix"
noarch::dask-cuda-21.8.0-pyhd8ed1ab_0.tar.bz2
-    "python >=3.7"
+    "python >=3.7",
+    "click <8.1",
+    "__unix"
noarch::dask-cuda-22.2.0-pyhd8ed1ab_0.tar.bz2
-    "python >=3.7"
+    "python >=3.7",
+    "click <8.1",
+    "__unix"
```

**Comment by jakirkham** [comment]
Since we are not constraining that elsewhere (like PyPI), think we should leave it. Otherwise it may lead to confusion.

**Comment by beckermr** [comment]
The pin here is rather specific. It should be avoided if possible.

**Comment by jakirkham** [comment]
In general yes, but this is what my colleagues and I agreed to as it is inline with other RAPIDS projects. Please see PR ( https://github.com/rapidsai/dask-cuda/pull/883 ) for more details

File: recipe/gen_patch_json.py [hunk 1]
```diff
@@ -1398,6 +1398,13 @@ def _gen_new_index_per_key(repodata, subdir, index_key):
             _replace_pin("flake8 >=3.5", "flake8 >=4.0", record["depends"], record)
             _replace_pin("pytest >=3.5", "pytest >=7.0", record["depends"], record)
 
+        # older versions of dask-cuda do not work on non-UNIX operating systems and must be constrained to UNIX
+        # issues in click 8.1.0 cause failures for older versions of dask-cuda
+        if record_name == "dask-cuda" and record.get("timestamp", 0) <= 1645130882435:  # 22.2.0 and prior
+            new_depends = record.get("depends", [])
+            new_depends += ["click ==8.0.4", "__unix"]
```
Comment (charlesbluca) [inline recipe/gen_patch_json.py:1405]: As @jakirkham suggested, we will probably want to constrain further to just Linux (i.e. excluding MacOS):

```suggestion
            new_depends += ["click ==8.0.4", "__linux"]
```

**Review by jakirkham — APPROVED** [review]

**Comment by jakirkham** [comment]
Thanks all! 😄

**Pull request merged**